server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -2124,6 +2126,15 @@ public static void toXContent(Metadata metadata, XContentBuilder builder, ToXCon
             builder.endObject();
 
             if (context == XContentContext.API) {
+                if (params.paramAsBoolean(MAPPINGS_BY_HASH_PARAM, true)) {
+                    builder.startObject("mappings");
```

Comment:
Perhaps we can call this `shared_mappings` and then have a flag of the same name to the API as well as a cluster setting to disable dumping non-shared mappings?